### PR TITLE
[loki-distributed] Add support for PSP

### PR DESCRIPTION
**Pull request by piontec**
Loki-distributed is missing PSP (which is present in non-distributed loki).

### CODE REVIEW

File: charts/loki-distributed/templates/role.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+kind: Role
+metadata:
+  name: {{ include "loki.fullname" . }}
+  labels:
+    {{- include "loki.labels" . | nindent 4 }}
+rules:
+- apiGroups:      ['extensions']
+  resources:      ['podsecuritypolicies']
+  verbs:          ['use']
+  resourceNames:  [{{ include "loki.fullname" . }}]
```
Comment (unguiculus) [inline charts/loki-distributed/templates/role.yaml:12]: Indent lists and use block style for lists.

File: charts/loki-distributed/templates/rolebinding.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  name: {{ include "loki.fullname" . }}
+  labels:
+    {{- include "loki.labels" . | nindent 4 }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: {{ include "loki.fullname" . }}
+subjects:
+- kind: ServiceAccount
+  name: {{ include "loki.serviceAccountName" . }}
```
Comment (unguiculus) [inline charts/loki-distributed/templates/rolebinding.yaml:14]:
```suggestion
  - kind: ServiceAccount
    name: {{ include "loki.serviceAccountName" . }}
```

File: charts/loki-distributed/values.yaml [hunk 3]
```diff
@@ -147,6 +147,11 @@ serviceAccount:
   # -- Annotations for the service account
   annotations: {}
 
+# RBAC configuration
+rbac: 
+  # -- should PodSecurityPolicy be used
+  pspEnabled: True
```
Comment (unguiculus) [inline charts/loki-distributed/values.yaml:153]:
```suggestion
  pspEnabled: true
```

File: charts/loki-distributed/values.yaml [hunk 4]
```diff
@@ -147,6 +147,11 @@ serviceAccount:
   # -- Annotations for the service account
   annotations: {}
 
+# RBAC configuration
+rbac: 
+  # -- should PodSecurityPolicy be used
```
Comment (unguiculus) [inline charts/loki-distributed/values.yaml:152]:
```suggestion
  # -- If enabled, a PodSecurityPolicy is created
```

File: charts/loki-distributed/templates/podsecuritypolicy.yaml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  hostPID: false
+  runAsUser:
+    rule: 'MustRunAsNonRoot'
+  seLinux:
+    rule: 'RunAsAny'
+  supplementalGroups:
+    rule: 'MustRunAs'
+    ranges:
+    - min: 1
+      max: 65535
```
Comment (unguiculus) [inline charts/loki-distributed/templates/podsecuritypolicy.yaml:27]:
```suggestion
      - min: 1
        max: 65535
```

File: charts/loki-distributed/templates/podsecuritypolicy.yaml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  fsGroup:
+    rule: 'MustRunAs'
+    ranges:
+    - min: 1
+      max: 65535
```
Comment (unguiculus) [inline charts/loki-distributed/templates/podsecuritypolicy.yaml:32]:
```suggestion
      - min: 1
        max: 65535
```

**Review by unguiculus — CHANGES_REQUESTED** [review]
Thanks. I made a few comments.

**Comment by piontec** [comment]
Hey!
Sorry for the long break, I'll try to clean it up

File: charts/loki-distributed/Chart.yaml [hunk 7]
```diff
@@ -3,7 +3,7 @@ name: loki-distributed
 description: Helm chart for Grafana Loki in microservices mode
 type: application
 appVersion: 2.1.0
-version: 0.22.1
+version: 0.22.3
```
Comment (unguiculus) [inline charts/loki-distributed/Chart.yaml:6]:
```suggestion
version: 0.23.0
```

File: charts/loki-distributed/templates/podsecuritypolicy.yaml [hunk 8]
```diff
@@ -0,0 +1,36 @@
+{{ if .Values.rbac.pspEnabled }}
```
Comment (unguiculus) [inline charts/loki-distributed/templates/podsecuritypolicy.yaml:1]:
```suggestion
{{- if .Values.rbac.pspEnabled }}
```

File: charts/loki-distributed/templates/role.yaml [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+    {{- include "loki.labels" . | nindent 4 }}
+rules:
+- apiGroups:      
+  - extensions
+  resources:      
+  - podsecuritypolicies
+  verbs:          
+  - use
+  resourceNames:  
+  - {{ include "loki.fullname" . }}
```
Comment (unguiculus) [inline charts/loki-distributed/templates/role.yaml:16]: Lists should be indented. API group should be `policy`.
```suggestion
  - apiGroups:      
      - policy
    resources:      
      - podsecuritypolicies
    verbs:          
      - use
    resourceNames:  
      - {{ include "loki.fullname" . }}
```

File: charts/loki-distributed/templates/rolebinding.yaml [hunk 10]
```diff
@@ -0,0 +1,15 @@
+{{ if .Values.rbac.pspEnabled }}
```
Comment (unguiculus) [inline charts/loki-distributed/templates/rolebinding.yaml:1]:
```suggestion
{{- if .Values.rbac.pspEnabled }}
```

File: charts/loki-distributed/templates/role.yaml [hunk 11]
```diff
@@ -0,0 +1,18 @@
+{{ if .Values.rbac.pspEnabled }}
```
Comment (unguiculus) [inline charts/loki-distributed/templates/role.yaml:1]:
```suggestion
{{- if .Values.rbac.pspEnabled }}
```

**Review by unguiculus — CHANGES_REQUESTED** [review]
I have a few more remarks.

**Comment by piontec** [comment]
Ping @unguiculus , can you please have a look again?

File: charts/loki-distributed/values.yaml [hunk 12]
```diff
@@ -158,6 +158,11 @@ serviceAccount:
   # -- Annotations for the service account
   annotations: {}
 
+# RBAC configuration
+rbac:
+  # -- If enabled, a PodSecurityPolicy is created
+  pspEnabled: true
```
Comment (unguiculus) [inline charts/loki-distributed/values.yaml:164]: Default should be false.

File: charts/loki-distributed/Chart.yaml [hunk 13]
```diff
@@ -3,7 +3,7 @@ name: loki-distributed
 description: Helm chart for Grafana Loki in microservices mode
 type: application
 appVersion: 2.1.0
-version: 0.24.0
+version: 0.24.1
```
Comment (unguiculus) [inline charts/loki-distributed/Chart.yaml:6]: Bump minor version.

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Review by unguiculus — APPROVED** [review]

**Pull request merged**